src/main/java/org/sagebionetworks/web/client/widget/entity/renderer/NbConvertPreviewWidget.java

```diff
@@ -75,8 +76,22 @@ public NbConvertPreviewWidget(
       friendlyMaxFileSize =
         gwt.getFriendlySize(HtmlPreviewWidget.MAX_HTML_FILE_SIZE, true);
     }
+    String repoUrl = synapseProperties.getSynapseProperty(REPO_SERVICE_URL_KEY);
+    String stack = stackFromRepoEndpoint(repoUrl);
     nbConvertEndpoint =
-      synapseProperties.getSynapseProperty(NBCONVERT_ENDPOINT_PROPERTY);
+      synapseProperties
+        .getSynapseProperty(NBCONVERT_ENDPOINT_PROPERTY)
+        .replaceAll("%s", stack);
```

Comment:
I had to verify that '%' does not have some special meaning in regex, and it LGTM!